### PR TITLE
Make HUP fetch the device type slug from the API

### PR DESCRIPTION
**Pull request by sradevski**
I haven't tested the entire script, but I did test the snippet for setting the SLUG, and it should be fine. I'll try to get it tested tomorrow, but it can be reviewed in the meantime (which is why is marked as draft)

See: https://www.flowdock.com/app/rulemotion/resin-devices/threads/Do15pyYgb59vcSbSGVHNBD737ki
Change-type: minor
Signed-off-by: Stevche Radevski <stevche@balena.io>

### CODE REVIEW

File: upgrade-2.x.sh [hunk 1]
```diff
@@ -997,6 +983,27 @@ UUID=$(jq -r '.uuid' $CONFIGJSON)
 API_ENDPOINT=$(jq -r '.apiEndpoint' $CONFIGJSON)
 DELTA_ENDPOINT=$(jq -r '.deltaEndpoint' $CONFIGJSON)
 
+FETCHED_SLUG=$(curl -H "Authorization: Bearer ${APIKEY}" --silent --retry 5 \
+"${API_ENDPOINT}/v6/device?\$select=is_of__device_type&\$expand=is_of__device_type(\$select=slug)&\$filter=uuid%20eq%20%27${UUID}%27" 2>/dev/null \
+| jq '.d[0].is_of__device_type[0].slug'
```
Comment (saintaardvark) [inline upgrade-2.x.sh:988]: This needs to be `jq -r` ; otherwise the slug is set to `"device"` (notice the quotes!) instead of `device`.

**Review by saintaardvark — COMMENTED** [review]
Hey @sradevski -- I've tested this out on a device registered in staging.  Once I fixed the `jq` call (as noted above), I was able to upgrade everything just fine.  For reference, the sequence was:

- Install a device in staging as `test-private`
- Modify its type to `raspberrypi3`
- Try current HUP from dashboard to verify it does not work as expected; copy arguments from that invocation
- Copy your script on to device (with `jq` fix)
- Invoke your script with the same arguments
- Your script works just fine, reboot
- Verify device continues to work well
- Run HUP again from dashboard to make sure that *future* upgrades will also work
- HUP runs well, reboot
- Verify device continues to work well

This is all :+1: for me, but I'll let @xginn8 review as well.

File: upgrade-2.x.sh [hunk 2]
```diff
@@ -997,6 +983,27 @@ UUID=$(jq -r '.uuid' $CONFIGJSON)
 API_ENDPOINT=$(jq -r '.apiEndpoint' $CONFIGJSON)
 DELTA_ENDPOINT=$(jq -r '.deltaEndpoint' $CONFIGJSON)
 
+FETCHED_SLUG=$(curl -H "Authorization: Bearer ${APIKEY}" --silent --retry 5 \
```
Comment (xginn8) [inline upgrade-2.x.sh:986]:
```suggestion
FETCHED_SLUG=$(CURL_CA_BUNDLE=${TMPCRT} curl -H "Authorization: Bearer ${APIKEY}" --silent --retry 5 \
```

Comment (xginn8) [inline upgrade-2.x.sh:986]:
```suggestion
FETCHED_SLUG=$(CURL_CA_BUNDLE=${TMPCRT} curl -H "Authorization: Bearer ${APIKEY}" --silent --retry 5 \
```

**Review by xginn8 — COMMENTED** [review]
lgtm otherwise

**Comment by sradevski** [comment]
@balena-ci I self-certify! since I addressed the Matt's change, I'll give this another test in staging and then we can deploy it to prod.

**Comment by sradevski** [comment]
@balena-ci I self-certify!

**Comment by sradevski** [comment]
@xginn8 how is balenahup "deployed"?

**Pull request merged**